node/src/components/small_network.rs

```diff
@@ -510,7 +509,17 @@ where
                     .add_incoming(peer_addr, Instant::now())
                 {
                     self.connection_completed(peer_id);
-                    self.update_joining_set(peer_id, is_joiner);
+
+                    // We should not update the joining set when we receive an incoming connection,
+                    // because the `message_sender` which is handling the corresponding outgoing
+                    // connection will not receive the update of the remote joiner state.
+                    //
+                    // Such desync may cause the node to try to send requests that are not safe for
+                    // joiners, to the joining node, because the outgoing connection may outlive the
+                    // incoming connection, i.e. it may take some time to drop "our" outgoing
+                    // connection after a peer has closed the corresponding incoming connection.
+
+                    // self.update_joining_set(peer_id, is_joiner);
```

Comment:
Is this left commented out on purpose?

Reply:
Yes, to make the way we attempted to fix the bug explicit. I see no problem removing it if you'd like to.

Reply:
I think our policy is to remove commented out code, so let's take it out!

Reply:
Removed in fd62aa2c41eed5489c3576b82438fe5d964603f2